File: Easy/13. Roman to Integer.py
# 13. Roman to Integer
class Solution(object):
     def romanToInt(self, s):
        sum = 0
        prevValue = 0
        values = [0] * 256
        values[ord('I')] = 1; values[ord('V')] = 5; values[ord('X')] = 10; values[ord('L')] = 50
        values[ord('C')] = 100; values[ord('D')] = 500; values[ord('M')] = 1000
        
        for c in s:
            currentValue = values[ord(c)]
            sum += (currentValue - 2 * prevValue) if (currentValue > prevValue) else currentValue
            prevValue = currentValue
        return sum

File: Easy/13. Roman to Integer.java
// 13. Roman to Integer
import java.util.HashMap;

class Solution {
    public int romanToInt(String s) {
        HashMap<Character, Integer> translations = new HashMap<>();
        translations.put('I', 1);
        translations.put('V', 5);
        translations.put('X', 10);
        translations.put('L', 50);
        translations.put('C', 100);
        translations.put('D', 500);
        translations.put('M', 1000);
        
        int number = 0;
        s = s.replace("IV", "IIII").replace("IX", "VIIII");
        s = s.replace("XL", "XXXX").replace("XC", "LXXXX");
        s = s.replace("CD", "CCCC").replace("CM", "DCCCC");
        
        for (char c : s.toCharArray()) {
            number += translations.get(c);
        }
        
        return number;
    }
}
